### PR TITLE
Add default nightly testing on JDK16

### PR DESCRIPTION
**Pull request by sxa**
Fixes https://github.com/AdoptOpenJDK/openjdk-build/issues/2337

NOTE: I can't do the same for JDK17 for now as the jobs haven't been created, but doing this one step at a time to check machine capacity is a good thing. We could also wait until JDK15 is no longer needed before enabling JDK17 fully

Notes to reviewers:
1. PLEASE be certain I haven't made any typos in the syntax
2. PLEASE leave it for me to merge so I can do it at a time I'm paying attention to the pipelines when it first gooes through

### CODE REVIEW

**Review by smlambert — APPROVED** [review]
thanks @sxa ! 

(and I have generated jdk17 test jobs as you have mentioned them in your description to be enabled when jdk15 is done)

File: pipelines/jobs/configurations/jdk16_pipeline_config.groovy [hunk 1]
```diff
@@ -68,10 +62,7 @@ class Config16 {
                 os                  : 'windows',
                 arch                : 'x64',
                 additionalNodeLabels: 'win2012&&vs2017',
-                test                : [
-                        nightly: [],
-                        weekly : ['sanity.openjdk', 'sanity.perf', 'sanity.system', 'extended.system']
-                ]
+                test                : 'default',
```
Comment (M-Davies) [inline pipelines/jobs/configurations/jdk16_pipeline_config.groovy:65]:
```suggestion
                test                : 'default'
```

**Review by M-Davies — APPROVED** [review]
Just one non blocker. LGTM

File: build-farm/platform-specific-configurations/windows.sh [hunk 2]
```diff
@@ -152,7 +152,7 @@ then
       export CONFIGURE_ARGS_FOR_ANY_PLATFORM="${CONFIGURE_ARGS_FOR_ANY_PLATFORM} --with-freemarker-jar=/cygdrive/c/openjdk/freemarker.jar"
     elif [ "${JAVA_TO_BUILD}" == "${JDK11_VERSION}" ]
     then
-      TOOLCHAIN_VERSION="2019"
+      TOOLCHAIN_VERSION="2017"
```
Comment (karianna) [inline build-farm/platform-specific-configurations/windows.sh:155]: Why the drop back to 2017?

**Review by karianna — CHANGES_REQUESTED** [review]

File: build-farm/platform-specific-configurations/windows.sh (hunk 2 above)
Comment (karianna) [reply build-farm/platform-specific-configurations/windows.sh:155]: check work so I'll let this through

**Review by karianna — APPROVED** [review]

**Pull request merged**